sylt-compiler/src/typechecker.rs

```diff
@@ -1132,6 +1222,13 @@ impl TypeChecker {
                     .map(|(name, ty)| (name.clone(), self.inner_bake_type(*ty, seen)))
                     .collect(),
             ),
+            Type::Enum(name, variants) => RuntimeType::Enum(
+                name.clone(),
+                variants
+                    .iter()
+                    .map(|(name, ty)| (name.clone(), self.inner_bake_type(*ty, seen)))
+                    .collect(),
+            ),
```

Comment:
Again, this might be able to be merged with blob.

Reply:
`RuntimeType::Enum` is different - I didn't feel this varented a function or a sub match-expression. I felt this read better.